autopush/websocket.py

```diff
@@ -352,13 +360,31 @@ def finish_hello(self, result):
             self.sendMessage(json.dumps(msg).encode('utf8'), False)
             return
 
+        if result.get("Attributes", {}).get("node_id"):
+            # Get the previous information returned from dynamodb.
+            attrs = result.get("Attributes")
+            node_id = self._get_aval(attrs.get("node_id", {}), 'S')
+            uaid = self._get_aval(attrs.get("uaid", {}), 'S')
+            connected_at = self._get_aval(attrs.get("connected_at", {}), 'N')
+            url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
+            if not getattr(self, 'testing', False):
+                d = self.ap_settings.agent.request(
+                    "DELETE",
+                    url.encode("utf8"),
+                ).addCallback(self.finish_hello)
+                d.addErrback(self.log_err, extra="Failed to delete old node")
+                return
+        self.finish_hello()
+
+    def finish_hello(self, *args):
+        self._register = None
         msg = {"messageType": "hello", "uaid": self.uaid, "status": 200}
         self.ap_settings.clients[self.uaid] = self
         self.sendJSON(msg)
         self.metrics.increment("updates.client.hello", tags=self.base_tags)
         self.process_notifications()
 
-    def process_notifications(self):
+    def process_notifications(self, *args):
```

Comment:
Ah, missed that one. Originally I was calling process_notifications from a defer and was tossing the arg because we didn't need it. Will remove.